pkg/apis/serving/v1beta1/predictor_torchserve.go

```diff
@@ -42,30 +43,46 @@ type TorchServeSpec struct {
 	PredictorExtensionSpec `json:",inline"`
 }
 
+var _ ComponentImplementation = &TorchServeSpec{}
+
 // Validate returns an error if invalid
 func (t *TorchServeSpec) Validate() error {
+	return utils.FirstNonNilError([]error{
+		validateStorageURI(t.GetStorageUri()),
+		t.validateGPU(),
+	})
+}
+
+func (t *TorchServeSpec) validateGPU() error {
 	if utils.IsGPUEnabled(t.Resources) && !strings.Contains(*t.RuntimeVersion, PyTorchServingGPUSuffix) {
 		return fmt.Errorf(InvalidPyTorchRuntimeIncludesGPU)
 	}
 
 	if !utils.IsGPUEnabled(t.Resources) && strings.Contains(*t.RuntimeVersion, PyTorchServingGPUSuffix) {
 		return fmt.Errorf(InvalidPyTorchRuntimeExcludesGPU)
 	}
-	return utils.FirstNonNilError([]error{
-		validateStorageURI(t.GetStorageUri()),
-	})
+	return nil
 }
 
 // Default sets defaults on the resource
 func (t *TorchServeSpec) Default(config *InferenceServicesConfig) {
 	t.Container.Name = constants.InferenceServiceContainerName
 	if t.RuntimeVersion == nil {
-		if utils.IsGPUEnabled(t.Resources) {
-			t.RuntimeVersion = proto.String(config.Predictors.PyTorch.DefaultGpuImageVersion)
+		if t.ProtocolVersion != nil && *t.ProtocolVersion == constants.ProtocolV2 {
+			if utils.IsGPUEnabled(t.Resources) {
+				t.RuntimeVersion = proto.String(config.Predictors.PyTorch.V2.DefaultGpuImageVersion)
+			} else {
+				t.RuntimeVersion = proto.String(config.Predictors.PyTorch.V2.DefaultImageVersion)
+			}
 		} else {
-			t.RuntimeVersion = proto.String(config.Predictors.PyTorch.DefaultImageVersion)
+			if utils.IsGPUEnabled(t.Resources) {
+				t.RuntimeVersion = proto.String(config.Predictors.PyTorch.V1.DefaultGpuImageVersion)
+			} else {
+				t.RuntimeVersion = proto.String(config.Predictors.PyTorch.V1.DefaultImageVersion)
+			}
 		}
 	}
+
 	if t.ModelClassName == "" {
```

Comment:
Add the protocolVersion check here? I think this field is not needed by torchserve correct?